commit 83f424c9feb291e491920357d4ad6949e767e169
Author: Aurélien Gâteau <agateau@kde.org>
Date:   Sun Mar 11 23:16:43 2012 +0100

    Correctly set DocumentViewContainer parent
    
    Fix GVPart usage in Ark
    BUG: 292394
    FIXED-IN: 2.8.2

diff --git a/part/gvpart.cpp b/part/gvpart.cpp
index d50879d..67de8c5 100644
--- a/part/gvpart.cpp
+++ b/part/gvpart.cpp
@@ -56,11 +56,11 @@ K_EXPORT_PLUGIN(GVPartFactory)
 namespace Gwenview
 {
 
-GVPart::GVPart(QWidget* /*parentWidget*/, QObject* parent, const QVariantList& /*args*/)
+GVPart::GVPart(QWidget* parentWidget, QObject* parent, const QVariantList& /*args*/)
 : KParts::ReadOnlyPart(parent)
 {
     KGlobal::locale()->insertCatalog("gwenview");
-    DocumentViewContainer* container = new DocumentViewContainer;
+    DocumentViewContainer* container = new DocumentViewContainer(parentWidget);
     setWidget(container);
     mDocumentView = container->createView();
 
